compiler/passes/errorHandling.cpp

```diff
@@ -1071,6 +1082,91 @@ void ErrorCheckingVisitor::exitDeferStmt(DeferStmt* node) {
 
 } /* end anon namespace */
 
+Symbol* findErrorVarForHandlerLabel(LabelSymbol* handlerLabel) {
+  // find the error that this handlerLabel is working with
+  for(Expr* e = handlerLabel->defPoint->next; e != NULL; e = e->next) {
+    std::vector<CallExpr*> calls;
+    collectCallExprs(e, calls);
+    for_vector(CallExpr, call, calls) {
+      if (call->isPrimitive(PRIM_CHECK_ERROR)) {
+        SymExpr* se = toSymExpr(call->get(1));
+        INT_ASSERT(se->symbol()->hasFlag(FLAG_ERROR_VARIABLE));
+        return se->symbol();
+      }
+    }
+  }
+
+  INT_FATAL("Could not find error variable for handler");
+  return NULL;
+}
+
+struct ErrorVarAndGoto { Symbol* errorVar; GotoStmt* gotoHandler; };
+
+static ArgSymbol* errorFormal(FnSymbol* fn) {
+  for_formals(formal, fn)
+    if (formal->hasFlag(FLAG_ERROR_VARIABLE))
+      return formal;
+  return NULL;
+}
+
+// This mimicks the control flow of a thrown error, for an IBB.
+// The label is created analogously to ErrorHandlingVisitor::enterForallStmt().
+// The goto looks like the one created in ErrorHandlingVisitor::gotoHandler().
+// This ignores try-blocks because an IBB is placed in an iterator to handle
+// the errors thrown in the loop body. Iterator try-blocks, if any, should not
+// be used to handle the loop body's errors.
+static ErrorVarAndGoto forallHandlerForIBB(ForallStmt* fs) {
```

Comment:
should it be named like `constructForallHandlerForIBB` or `create` or `add` or ... ?

Reply:
Oh it seems maybe you chose this name because it is idempotent? Needs to be clearer in the comment above it.